### PR TITLE
Improve docker support

### PR DESCRIPTION
**Pull request by crash7**
Hello! I was having some problems trying to test this on my machine using docker and I had improved the setup a little. 

Main changes:
- Add docker-compose file
- Add environment variable to initialize the database (just change the value of INIT_DATABASE to "no" to disable it) 
- Add custom entrypoint that generates the .env file based on the environment variables (defined in docker-compose)
- Update Dockerfile (gettext-base package)

This should help solve https://github.com/firefly-iii/firefly-iii/issues/537

I can add more documentation if needed!

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - FF_APP_KEY=SomeRandomStringOf32CharsExactly
+      - FF_APP_ENV=development
+    ports:
+      - "80:80"
+    links:
+      - firefly-db
+
+volumes:
+  firefly-storage:
+    driver: local
```
Comment (steelbrain) [inline docker-compose.yml]: EOF file

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	# depends on your machine, but it may take a file to boot mysql container the first time
+	until php artisan firefly:verify &>/dev/null
+	do
+  		echo "waiting mysql"
+  		sleep 10 
+	done
+	php artisan migrate:refresh --seed
+fi
+
+exec apache2-foreground
```
Comment (steelbrain) [inline docker/entrypoint.sh]: Similarly, EOF here as well please

File: docker-compose.yml (hunk 1 above)
Comment (JC5) [inline docker-compose.yml]: Done

File: docker/entrypoint.sh (hunk 2 above)
Comment (JC5) [inline docker/entrypoint.sh]: Done

**Comment by JC5** [comment]
Merged!

**Comment by crash7** [comment]
Thanks! And thanks @steelbrain for the review, I forgot the EOF since I normally use editor config and it handles this for me.

**Pull request merged**